internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -198,12 +199,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 
 // reconcile handles cluster reconciliation.
 func (r *Reconciler) reconcile(ctx context.Context, s *scope.Scope) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
 	var err error
 
 	// Get ClusterClass.
 	clusterClass := &clusterv1.ClusterClass{}
 	key := client.ObjectKey{Name: s.Current.Cluster.Spec.Topology.Class, Namespace: s.Current.Cluster.Namespace}
 	if err := r.Client.Get(ctx, key, clusterClass); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(6).Info("could not find ClusterClass", "ClusterClass", klog.KRef(key.Namespace, key.Name))
```

Comment:
Sorry - one more nit I missed. Let's set the logging value here to 2. We shouldn't use logging levels higher than 5 according to the logging guidelines: https://cluster-api.sigs.k8s.io/developer/logging.html#log-levels

IMO this is good information for users about what's happening, and it keeps the logging similar to what happens today as returned reconciler errors are always logged.

Reply:
Just wondering if 2 is too spamy considering we actually don't do anything. Maybe let's go with 3, wdyt?

Reply:
Given we don't requeue this isn't that spammy, and today this is always logged. I'd default to 2 because this is a substantial error. 

We also lose a TopologyReconciled error condition when removing this error AFAIK which might make it more difficult for users to discover what's going wrong.

Reply:
OK. 

> We shouldn't use logging levels higher than 5 

https://github.com/kubernetes-sigs/cluster-api/blob/main/controllers/remote/cluster_cache_tracker.go#L543

But I see that there are also 6 levels being used here. 
Do I need to make the modifications together?

Reply:
This will be logged once for every update on the Cluster, which can be very spamy. I would argue even more spamy then before. Because before we had an exponential backoff with the error

Reply:
The more I think about it, the more I wonder if the previous behavior was better. Losing the condition is very very bad